src/Components/OATHeader/internal/FileSubMenu.tsx

```diff
@@ -0,0 +1,178 @@
+import React, { useState, useEffect } from 'react';
+import { ActionButton, Text, Callout } from '@fluentui/react';
+import { useTranslation } from 'react-i18next';
+import { getSubMenuItemStyles, getSubMenuStyles } from '../OATHeader.styles';
+import {
+    OATDataStorageKey,
+    OATFilesStorageKey
+} from '../../../Models/Constants/Constants';
+import { IAction } from '../../../Models/Constants';
+import { IOATEditorState } from '../../../Pages/OATEditorPage/OATEditorPage.types';
+import { SET_OAT_PROJECT_NAME } from '../../../Models/Constants/ActionTypes';
+
+type IFileSubMenu = {
+    dispatch?: React.Dispatch<React.SetStateAction<IAction>>;
+    resetProject?: () => void;
+    setModalBody?: React.Dispatch<React.SetStateAction<string>>;
+    setModalOpen?: React.Dispatch<React.SetStateAction<boolean>>;
+    setSubMenuActive: React.Dispatch<React.SetStateAction<boolean>>;
+    subMenuActive?: boolean;
+    state?: IOATEditorState;
+    targetId?: string;
+};
+
+export const FileSubMenu = ({
+    dispatch,
+    resetProject,
+    setModalBody,
+    setModalOpen,
+    setSubMenuActive,
+    subMenuActive,
+    state,
+    targetId
+}: IFileSubMenu) => {
+    const { t } = useTranslation();
+    const subMenuItemStyles = getSubMenuItemStyles();
+    const subMenuStyles = getSubMenuStyles();
+    const [files] = useState(
+        JSON.parse(localStorage.getItem(OATFilesStorageKey))
+    );
+    const [isFileStored, setIsFileStored] = useState(false);
+    const [fileIndex, setFileIndex] = useState(-1);
+    const { projectName } = state;
+
+    const handleSave = () => {
+        setSubMenuActive(false);
+
+        if (isFileStored) {
+            // Update file
+            const editorData = JSON.parse(
+                localStorage.getItem(OATDataStorageKey)
+            );
+            files[fileIndex].data = editorData;
+            localStorage.setItem(OATFilesStorageKey, JSON.stringify(files));
```

Comment:
I think it would be useful to encapsulate the localStorage code into some kind of service class.

I.e. so we can expose more meaningful operations to save/load etc.